src/current/v23.1/changefeed-messages.md

```diff
@@ -264,7 +266,7 @@ As an example, you run the following sequence of SQL statements to create a chan
     {"after": {"id": 4, "name": "Danny", "office": "los angeles"}, "key": [4], "updated": "1701102561022789676.0000000000"}
     ~~~
 
-    The messages received at the sink are in order by timestamp **for each key**. Here, the update for key `[1]` is emitted before the insertion of key `[2]` even though the timestamp for the update to key `[1]` is higher. That is, if you follow the sequence of updates for a particular key at the sink, they will always be in the correct timestamp order.
+    The messages received at the sink are in order by timestamp **for each key**. Here, the update for key `[1]` is emitted before the insertion of key `[2]` even though the timestamp for the update to key `[1]` is higher. That is, if you follow the sequence of updates for a particular key at the sink, they will be in the correct timestamp order. However, re-emitted messages (duplicates) may not be in the correct timestamp order. For details on when changefeeds might re-emit messages, refer to [Duplicate messages](#duplicate-messages).
```

Comment:
> However, re-emitted messages (duplicates) may not be in the correct timestamp order

I thought it was more like "re-emitted messages may not contain all updates after the closed timestamp." Or something like that. Basically, if its A, B, C, D then we will re-emit C at the right mvcc timestamp, but we may not re-emit D.

Reply:
I believe your first question is yes and the second is no. That is, not everything will be re-emitted but what is re-emitted won't be out of order. @rharding6373 can you confirm?

Reply:
Andy is correct based on our current implementation, but I think it would be confusing to describe a scenario where we have duplicates emitted A B C B D, which is valid, as in order. @awoods187 do you see a reason why we shouldn't just say that duplicates don't follow the ordering and delivery guarantees?

Reply:
@awoods187 the question I think is do we want to put duplicates will be emitted in order in our contract now, or leave this open until we figure out what guarantees we want to maintain? We don't have any changes on the roadmap that would potentially change this behavior afaik, but I want to be sure we don't shoot ourselves in the foot by including this implementation detail.

Reply:
I think we want to leave it open until we make a product decision. So I think we would say duplicates don't follow the ordering and delivery guarantees.